api/src/main/java/jakarta/persistence/Query.java

```diff
@@ -36,6 +55,32 @@
  * @since 1.0
  */
 public interface Query {
+    /**
+     * Obtain a {@link TypedQuery} with the given query result type,
+     * which must be a supertype of the result type of this query.
+     * This query must be a Jakarta Persistence {@code SELECT} query
+     * or a native SQL query which returns a result set.
+     * @param resultType The Java class of the query result type
+     * @param <R> The query result type
+     * @throws IllegalArgumentException if the given result type is
+     *         not a supertype of the result type of this query
+     * @since 4.0
+     */
+    <R> TypedQuery<R> ofType(Class<R> resultType);
```

Comment:
Who prefers `withResultType()` here?

Reply:
I would say either is fine.  `withResultType` is more precise.  `ofType` works because the method transforms a `Query` into a `TypedQuery`.